pkg/appmanager/appManager_test.go

```diff
@@ -3443,6 +3443,176 @@ var _ = Describe("AppManager Tests", func() {
 					flatDg = nsMap.FlattenNamespaces()
 					Expect(flatDg).To(BeNil(), "should not have data")
 				})
+
+				It("manages alternate backends for routes (pool creation)", func() {
+					svc1Name := "foo"
+					svc2Name := "bar"
+					svc3Name := "baz"
+					fooSvc := test.NewService(svc1Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37001}})
+					mockMgr.addService(fooSvc)
+					barSvc := test.NewService(svc2Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37002}})
+					mockMgr.addService(barSvc)
+					bazSvc := test.NewService(svc3Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37003}})
+					mockMgr.addService(bazSvc)
+
+					spec := routeapi.RouteSpec{
+						Host: "foo.com",
+						Path: "/foo",
+						To: routeapi.RouteTargetReference{
+							Kind: "Service",
+							Name: "fake1", // start with non-existent service
+						},
+						AlternateBackends: []routeapi.RouteTargetReference{
+							{
+								Kind: "Service",
+								Name: "fake2", // start with non-existent service
+							},
+							{
+								Kind: "Service",
+								Name: svc3Name,
+							},
+						},
+					}
+					route := test.NewRoute("route", "1", namespace, spec, nil)
+					mockMgr.addRoute(route)
+
+					resources := mockMgr.resources()
+					rs, ok := resources.Get(
+						serviceKey{svc3Name, 80, "default"}, "ose-vserver")
+					Expect(ok).To(BeTrue(), "Route should be accessible.")
+					Expect(rs).ToNot(BeNil(), "Route should be object.")
+
+					Expect(resources.PoolCount()).To(Equal(1))
+					Expect(len(rs.Policies)).To(Equal(0))
+
+					// Update to real service
+					route.Spec.To.Name = svc1Name
+					route.Spec.AlternateBackends[0].Name = svc2Name
+					mockMgr.updateRoute(route)
+					Expect(resources.PoolCount()).To(Equal(3))
+					rs, _ = resources.Get(
+						serviceKey{svc1Name, 80, "default"}, "ose-vserver")
+					Expect(len(rs.Policies)).To(Equal(0))
+
+					// Remove an alternate service
+					mockMgr.deleteService(bazSvc)
+					Expect(resources.PoolCount()).To(Equal(2))
+
+					mockMgr.addService(bazSvc)
+					Expect(resources.PoolCount()).To(Equal(3))
+				})
+
+				It("manages alternate backends for routes (datagroups)", func() {
+					svc1Name := "foo"
+					svc2Name := "bar"
+					svc3Name := "baz"
+					fooSvc := test.NewService(svc1Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37001}})
+					mockMgr.addService(fooSvc)
+					barSvc := test.NewService(svc2Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37002}})
+					mockMgr.addService(barSvc)
+					bazSvc := test.NewService(svc3Name, "1", namespace, "NodePort",
+						[]v1.ServicePort{{Port: 80, NodePort: 37003}})
+					mockMgr.addService(bazSvc)
+
+					wMax := int32(100)
```

Comment:
Nit - not sure why this is called wMax.  100 isn't the max (I think 256 is).  In anycase, this seems to be the weight of the foo service, so that would be a better name (or w# like you have for the other weights).